### ISSUE TITLE
Refactor MarathonScheduler.newTask

### ISSUE THREAD

**Issue by ConnorDoyle** [mesosphere/marathon#937]
There is a long-standing TODO in the in-line comments to fix the return value of this method.  It currently returns an anonymous tuple wrapped in an option.

**Comment by hekaldama** [comment]
The TODO comment was removed in #936 (646e0fd300935) but the optimization / cleanup remains.

**Comment by hekaldama** [comment]
@ConnorDoyle I found https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala#L500-L516 as well. Does that need to be removed? Doesn't seem like anything is calling it, but not sure if I am understanding.

**Comment by kolloch** [comment]
TODO has been removed.

**Comment by kolloch** [comment]
I just noticed that the method was gone so that the issue text didn't make sense anymore. That was a mistake. Sorry. We are a bit overwhelmed with the long issue list and tried to remove the issues that are already done. Thanks for getting back to us about that.

I created a PR for this. I would appreciate your review since you brought this issue up.

Generally, I think it is not helpful creating issues for internal style issues which are that easily fixed without following up very soon. Otherwise our issues list will grow very long and that results in accumulated overhead in planning. 

But I know that one can easily disagree about this.
